evap/static/ts/src/sisyphus.ts

```diff
@@ -0,0 +1,312 @@
+     * Restore form field data from local storage
+     */
+    restoreFieldsData(field: Element, resque: string) {
+        if (field.getAttribute("name") === undefined) {
+            return false;
+        }
+        if (
+            field instanceof HTMLInputElement &&
+            field.type === "checkbox" &&
+            resque !== "false" &&
+            !field.name.includes("[")
+        ) {
+            field.checked = true;
+        } else if (
+            field instanceof HTMLInputElement &&
+            field.type === "checkbox" &&
+            resque === "false" &&
+            !field.name.includes("[")
+        ) {
+            field.checked = false;
+        } else if (field instanceof HTMLInputElement && field.type === "radio") {
+            if (field.value === resque) {
+                field.checked = true;
+            }
+        } else if (field instanceof HTMLInputElement && !field.name.includes("[")) {
+            field.value = resque;
+        } else {
+            // @ts-expect-error Definitely an input field with a value, but not known by type
+            field.value = resque.split(",");
+        }
+    }
+
+    /**
+     * Bind immediate saving (on typing/checking/changing) field data to local storage when user fills it
+     */
+    bindSaveDataImmediately(field: HTMLInputElement | HTMLTextAreaElement, prefix: string) {
+        field.addEventListener("input", () => {
+            this.saveToBrowserStorage(prefix, field.value);
+        });
+    }
+
+    /**
+     * Save data to Local Storage and fire callback if defined
+     */
+    saveToBrowserStorage(key: string, value: any, fireCallback?: boolean) {
+        // if fireCallback is undefined it should be true
+        fireCallback = fireCallback ?? true;
+        // eslint-disable-next-line @typescript-eslint/restrict-plus-operands
+        localStorage.setItem(key, value + "");
+        if (fireCallback && value !== "") {
+            this.options.onSave(this);
+        }
+    }
+
+    /**
+     * Bind saving field data on change
+     */
+    bindSaveDataOnChange(field: Element) {
+        field.addEventListener("change", () => {
+            this.saveAllData();
+        });
+    }
+
+    /**
+     * Saving (by timeout) field data to local storage when user fills it
+     */
+    saveDataByTimeout() {
+        setTimeout(() => {
+            const timeout = () => {
+                this.saveAllData();
+                setTimeout(timeout, this.options.timeout * 1000);
+            };
+
+            return timeout;
+        }, this.options.timeout * 1000);
+    }
+
+    /**
+     * Bind release form fields data from local storage on submit/reset form
+     */
+    bindReleaseData() {
+        for (const target of this.targets) {
+            const releaseHandler = () => {
+                this.releaseData(target);
+            };
+            target.addEventListener("submit", releaseHandler);
+            target.addEventListener("reset", releaseHandler);
```

Comment:
... but even without `autoRelease`, we might want to have this event handler. We don't have any `type="reset"` inputs in EvaP, so it would be fine to ignore it. What do you think?